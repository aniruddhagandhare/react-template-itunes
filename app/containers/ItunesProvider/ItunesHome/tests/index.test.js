/**
 *
 * Tests for ItunesHome
 *
 *
 */

 import React from 'react';
 import { renderProvider, timeout } from '@utils/testUtils';
 import { fireEvent } from '@testing-library/dom'
 import { ItunesHomeTest as ItunesHome } from '../index';
 
 describe('<ItunesHome /> container tests', () => {
   let submitSpy
 
   beforeEach(() => {
     submitSpy = jest.fn()
   });
   it('should render and match the snapshot', () => {
     const { baseElement } = renderProvider(<ItunesHome />);
     expect(baseElement).toMatchSnapshot();
   });

   it('should call dispatchGetSongs after input has changed', async () => {
       const requestGetSongs = jest.fn();
       const { getByTestId } = renderProvider(<ItunesHome dispatchGetSongs={requestGetSongs} />)
        fireEvent.change(getByTestId('search-term'), {
            target: { value: 'drake'}
        })
        await timeout(200);
        expect(requestGetSongs).toBeCalled();
   })

   it('should call dispatchClearSongs after the input is empty', async () => {
       const requestClearSongs = jest.fn();
       const requestGetSongs = jest.fn();
       const {getByTestId} = renderProvider(<ItunesHome dispatchGetSongs={requestGetSongs} dispatchClearSongs={requestClearSongs} />)
       fireEvent.change(getByTestId('search-term'), {
           target: { value: 'drake' }
       })
       await timeout(200);
       expect(requestGetSongs).toBeCalled();
       fireEvent.change(getByTestId('search-term'), {
           target: {value: ''}
       })
       await timeout(200);
       expect(requestClearSongs).toBeCalled();
   })
 });
 