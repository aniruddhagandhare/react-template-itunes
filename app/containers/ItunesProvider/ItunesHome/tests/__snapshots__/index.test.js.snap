// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItunesHome /> container tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ItunesHome__CenteredDiv-o6ln02-2 bIlxZJ"
    >
      <div
        class="ant-card ant-card-bordered"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              <p
                class="T__StyledText-gjlic1-0 cbAvsY"
                data-testid="t"
              >
                Search for your favourite tracks
              </p>
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ItunesHome__FlexWrapper-o6ln02-1 gRKyOl"
          >
            <input
              class="ant-input ItunesHome__CustomInput-o6ln02-0 cUUiMi"
              data-testid="search-term"
              placeholder="Search itunes"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="TrackGrid__Container-sc-1fqyrr9-4 glDrzJ"
        data-testid="track-grid"
      />
    </div>
  </div>
</body>
`;
